CONTRIBUTING.md

```diff
@@ -18,45 +18,163 @@ any additional information for contributing code to this project.
 
 ## Building
 
-### Egg format
+### Static/portable CLI binary
 
 ```
-$ ./bin/build
+$ ./bin/build_binary
 ```
 
-### Static/portable CLI binary
+### Egg format
 
 ```
-$ ./bin/build_binary
+$ ./bin/build
 ```
 
 ## Development
 
-- Create a directory that will hold all the virtualenv packages and files:
+To setup a development environment follow the instructions in this section. Once you have done so, you will be able to see live changes made to the CLI.
+
+1. Create a directory that will hold all the virtualenv packages and files. Note that you will only need to run this command once.
+
+macOS:
+
 ```
 $ python3 -m venv venv
 ```
 
-- Enable your terminal to use those files with this command:
+Windows:
+
+```
+py -m venv venv
+```
+
+2. Enable your terminal to use those files with this command. Note that this command will need to run each time you want to return to your virtual environment.
+
+macOS:
+
 ```
 $ source venv/bin/activate
 ```
 
-- Install requirements:
+Windows:
+
+```
+venv\Scripts\activate.bat
+```
+
+3. Install requirements
+
 ```
 $ pip3 install -r requirements.txt
 ```
 
-You can now run the tests and the CLI with modifiable files!
+4. You can now run the tests and the CLI with modifiable files.
+
+Check it out! Run the following prefix to start seeing changes
+
+```
+$ ./pkg_bin/conjur <command> <subcommand>
+```
 
 ## Testing
 
 ### Unit and Integration tests
 
+To run both unit and integration tests at once in a containerized environment:
+
 ```
 $ ./bin/test
 ```
 
+To run unit tests:
+```
+./bin/test_unit
+```
+
+You can run integration tests in the following ways:
+1. In a containerized environment (Python required)
+
+1. As a process (no Python required)
+
+#### Running in a containerized environment
+
+To run specific tests, perform the following:
+
+1. Drop down into a test container
+
+```
+$ ./bin/test_integration -d
```

Comment:
what's `-d`? do we have a full-name flag?

Reply:
-d is debug which brings up containers and throws us into the bash
```
if [ "$1" == "-d" ]; then
  DEBUG="true"
  shift
fi

...

if [[ "$DEBUG" == "true" ]]; then
  docker-compose exec test bash
  exit
fi
```

should I add a comment or allow for the `--debug` form as well?

Reply:
will add `if [[ "$1" = "--debug" || "$1" = "-d" ]]; then`

Reply:
i think that's the best solution. in the example we should use the full-name